homeassistant/components/notify/__init__.py

```diff
@@ -55,7 +57,8 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     hass.config.components.add(DOMAIN)
 
     if platform_setups:
-        await asyncio.wait([asyncio.create_task(setup) for setup in platform_setups])
+        with async_pause_setup(hass, SetupPhases.WAIT_PLATFORM_INTEGRATION):
+            await asyncio.wait([create_eager_task(setup) for setup in platform_setups])
```

Comment:
entity component creates tasks and does not wait on them.

all the legacy platforms (`device_tracker`, `notify`, `stt`, `tts`) wait in the top level integration for the legacy platforms. 

I'm not sure if we need to do that, but I didn't want to change that here because it would be a change in semantics

Reply:
Experiment with the device_tracker platform
https://github.com/home-assistant/core/pull/113715

If that works I think we can do the same for `notify`, `stt`, `tts`, and `scene`